maro/simulator/scenarios/oncall_routing/business_engine.py

```diff
@@ -44,6 +44,29 @@
 """
 
 
+def _get_staying_time(tick: int, plan: List[PlanElement], order_transition: DottableDict) -> int:
+    cur_coord = plan[0].order.coord
+    copied_orders: List[Order] = []
+    for i in range(len(plan)):
+        if plan[i].order.coord == cur_coord:
+            copied_orders.append(clone(plan[i].order))
+        else:
+            break
+
+    cur_tick = tick
+    processing_time = 0
+    for order in copied_orders:
+        order_status = order.get_status(cur_tick, order_transition)
+        if order_status != OrderStatus.DUMMY:
+            if order_status == OrderStatus.NOT_READY:
+                cur_tick = order.open_time - order_transition.buffer_before_open_time
+                processing_time = 0  # TODO
+            if order_transition.processing_proportion_to_quantity:
+                processing_time += order_transition.processing_time
+
```

Comment:
if processing_order and not self._config.order_transition.processing_proportion_to_quantity:
            processing_time = self._config.order_transition.processing_time